src/hyperion/external_interaction/callbacks/rotation/ispyb_callback.py

```diff
@@ -129,14 +130,12 @@ def activity_gated_start(self, doc: RunStart):
     def populate_info_for_update(
         self, event_sourced_data_collection_info: DataCollectionInfo, params
     ) -> Sequence[ScanDataInfo]:
+        assert (
+            self.ispyb_ids.data_collection_ids
+        ), "Expect an existing DataCollection to update"
         params = cast(RotationInternalParameters, params)
-        initial_collection_info = populate_data_collection_info_for_rotation(
-            params.hyperion_params.ispyb_params,
-            params.hyperion_params.detector_params,
-            params,
-        )
-        initial_collection_info = replace(
-            initial_collection_info,
+        data_collection_info = replace(
```

Comment:
Should: Is this now taking an empty data collection and recreating it from the existing `event_sourced_data_collection_info`? Why can't we just use the `event_sourced_data_collection_info` itself?

Reply:
Similar in the `ispyb_callback` for xray centring.

Reply:
yes, I should have spotted this!